Update install to refer to latest RabbitMQ (#1091)
StackStorm/st2docs: docs/source/upgrade_notes.rst

|st2| v3.6
----------

> * Prior to v3.5 then the installation instructions for all OSes except for CentOS/RHEL 8 were to use the version of RabbitMQ available in the OS distribution. This version is very old, and for 3.6 the installation instructions and simple install have been modified to install the latest version of RabbitMQ. It is not a requirement to upgrade RabbitMQ for installation of 3.6, but for non CentOS/RHEL 8 systems then to keep compatability with a clean installation then RabbitMQ cluster should be upgraded.

Reviewer: ```suggestion
* Prior to v3.5 the installation instructions for all OSes except for CentOS/RHEL 8
  said to use the version of RabbitMQ available in the OS distribution. This version is
  very old, and for 3.6 the installation instructions--and simple install--have been modified
  to install the latest version of RabbitMQ. It is not a requirement to upgrade RabbitMQ
  for installation of 3.6, but to keep compatibility with a clean installation, the RabbitMQ
  cluster should be upgraded for non CentOS/RHEL 8 systems.
```
